### PR TITLE
[Reviewer: Ellie] Add cassandra upgrade plugins

### PR DESCRIPTION
**Pull request by sebrexmetaswitch**
The Cassandra version is stored in etcd and managed by the
CassandraVersionPlugin

The CassandraUpgradePlugin is a queue manager plugin to handle the actual
upgrade. It does this by calling out to ```/usr/share/clearwater/clearwater-queue-manager/scripts/upgrade_cassandra.py``` if it exists

**Testing done**
 - UTs
 - Live tested
 - A fuller test plan and more extensive testing to follow

### CODE REVIEW

File: clearwater_queue_manager/cassandra_upgrade_plugin.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+from metaswitch.clearwater.queue_manager.plugin_base import QueuePluginBase
+from metaswitch.clearwater.etcd_shared.plugin_utils import run_command, safely_write
```
Comment (eleanor-merry) [inline clearwater_queue_manager/cassandra_upgrade_plugin.py:34]: Does this pass `make verify` in the clearwater-etcd build? I'd have thought it wouldn't as you've got unused imports and variables?

File: clearwater_queue_manager/cassandra_upgrade_plugin.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import logging
+import os
+import imp
+
+PACKAGES = "cassandra clearwater-cassandra homestead-cassandra homestead-prov-cassandra homer-cassandra memento-cassandra"
```
Comment (eleanor-merry) [inline clearwater_queue_manager/cassandra_upgrade_plugin.py:39]: You don't use most of these variables. Also, I'm not sure you should hardcode the packages names here - grab me to talk about this.

File: clearwater_queue_manager/cassandra_upgrade_plugin.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        self._wait_plugin_complete = params.wait_plugin_complete
+
+    def key(self):  # pragma: no cover
+        return "cassandra_upgrade"
+
+    def at_front_of_queue(self):
+        if os.path.exists(UPGRADE_SCRIPT):
+            cass_upgrade = imp.load_source("cass_upgrade", UPGRADE_SCRIPT)
+            result = cass_upgrade.do_upgrade()
+            if result:
```
Comment (eleanor-merry) [inline clearwater_queue_manager/cassandra_upgrade_plugin.py:57]: I'm confused by these checks - can we go through these?

File: clearwater_queue_manager/cassandra_upgrade_plugin.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import imp
+
+PACKAGES = "cassandra clearwater-cassandra homestead-cassandra homestead-prov-cassandra homer-cassandra memento-cassandra"
+SOURCES_FILE = "/etc/apt/sources.list.d/clearwater.list"
+UPGRADE_SCRIPT = "/usr/share/clearwater/clearwater-queue-manager/scripts/upgrade_cassandra.py"
+VERSION_FILE = "/etc/clearwater/cassandra_version"
+
+_log = logging.getLogger("cassandra_upgrade_plugin")
+
+class CassandraUpgradePlugin(QueuePluginBase):
```
Comment (eleanor-merry) [inline clearwater_queue_manager/cassandra_upgrade_plugin.py:46]: You'll want to set appropriate timeouts and alarms for this plugin - e.g. https://github.com/Metaswitch/clearwater-etcd-plugins/blob/split-clusters/chronos/apply_chronos_gr_config_plugin.py#L44

**Review by eleanor-merry — COMMENTED** [review]
Comments inline - I'm not sure your checks are quite right

**Comment by sebrexmetaswitch** [comment]
Not going to be used - closing

**Pull request closed without merging**